### ISSUE TITLE
Dark theme

### ISSUE THREAD

**Issue by septatrix** [observablehq/feedback#85]
**Is your feature request related to a problem? Please describe.**
Coding in the late hours is quite common for many people.
A dark theme would be very much appreciated to reduce eye strain.

**Describe the solution you'd like**
A theme chooser or automatic theme based on the `prefers-color-scheme` media query.
This should cover the notebook but also other web pages.

**Describe alternatives you've considered**
A further effort could be made to support different syntax highlighting schemes for the code editor.

**Comment by mootari** [comment]
Your request seems ambiguous to me:
- Are you requesting a global dark theme for observablehq.com? Observable [explicitely removed](https://talk.observablehq.com/t/feature-unannouncement-dark-mode/1440) its dark theme several years ago, and notebook authors cannot be expected to design for both light and dark.
- Or do you only want alternative color schemes for the editor? Be aware that you can already customize the editor scheme via user styles. See [this post](https://talk.observablehq.com/t/allow-users-to-create-notebook-themes/635/18) for an example.

**Comment by septatrix** [comment]
My request covers both. Regarding the editor I do not think user styles are a valid solution. Not every user wants nor does every browser support installing extensions. And even for those who use such an extension it is often a subpar user experience.

Concerning the global theme I am sad that the beta trial got removed. At that time I have not yet used observable so am unable to comment on how it looked. However it is a pity that the main reason it got removed (having to design for both themes) was already acknowledged. At that point the consensus seemed to be that (rightfully) no one is forced to design their notebooks for both themes.

Also some users have pointed to other projects which are able to handle this using CSS filter or other means. Furthermore would it be sufficient for most cases to have a white background per default on image, svg and canvas elements - and for the rare occurrences where this is not sufficient there are multiple workbooks which show easy ways to achieve this. This is all given that the author wants to go the extra mile to support dark themes which he obviously is not forced to (currentColor, manual background etc.).

A second point I would like to address is accessibility. Dark mode is not only a pure choice of taste but for some people also hugely beneficial for reading and viewing purposes. A carefully crafted official theme is often way better than some user styles...

**Comment by mootari** [comment]
Regardless of the actual effort of implementing and maintaining a dark theme by Observable, I believe that this would only be viable if notebook backgrounds still defaulted to white:
- Many notebooks (mine included) customize font colors which would completely break if forced onto a dark background.
- You can already adapt your notebook by injecting a style element with a media query for `prefers-color-scheme`.

**Comment by septatrix** [comment]
Yes I would be fine with a white background on the output cells. The injection of styles into ones notebooks on the other hand seems like a bad practice to me (especially when changing colors). On the one hand you are forcing color which might even override viewers user scripts (from extensions) and many people do not know a lot about contrast requirements. On the other hand this would also not solve the issue of a missing dark theme as this obviously would not transfer to when I am viewing other peoples creations.
